### ISSUE TITLE
Warn about undefined providers during bootstrap when there is a request scope dependency

### ISSUE THREAD

**Issue by Boshen** [nestjs/nest#4070]
## Feature Request

During bootstrap, when there exist a request scoped dependency somewhere deep in the application, we will get an undefined provider, and a nasty error is thrown, e.g.
```
TypeError: Cannot read property 'xxx' of undefined
    at FooModule.onApplicationBootstrap
```

I've encountered this numerous times, and each time it took me a while to realize that I'm injecting a request scoped dependency.

It would be much appreciated if nest can detect this and warn about it.

**Comment by kamilmysliwiec** [comment]
Please, provide a minimal repository which reproduces your issue.

**Comment by Boshen** [comment]
@kamilmysliwiec Here you go https://github.com/boshen/nestjs-playground

here's the relevant code: 
src/app.service.ts
```typescript
import { Injectable, Inject } from '@nestjs/common'
import { REQUEST } from '@nestjs/core'

@Injectable()
export class AppService {
  constructor(@Inject(REQUEST) private readonly req: any) {} // injected REQUEST

  getHello(): string {
    return 'Hello World!'
  }
}
```

src/app.module.ts
```typescript
import { Module, OnApplicationBootstrap } from '@nestjs/common'
import { AppController } from './app.controller'
import { AppService } from './app.service'

@Module({
  controllers: [AppController],
  providers: [AppService],
})
export class AppModule implements OnApplicationBootstrap {
  constructor(private readonly appService: AppService) {}

  onApplicationBootstrap() {
    console.log(this.appService) // undefined, can we warn about this?
  }
}
```

**Comment by Boshen** [comment]
Other people encountered this too, https://github.com/nestjs/bull/issues/159#issuecomment-523337734

**Comment by kamilmysliwiec** [comment]
This should be fixed in 6.11.8. Note that your lifecycle hooks won't be triggered in request-scoped classes.
